### PR TITLE
feat: complete some TODO guides

### PR DESCRIPTION
**Pull request by FC5570**

### CODE REVIEW

File: docs/Guide/arguments/creating-your-own-arguments.mdx [hunk 1]
```diff
@@ -3,4 +3,94 @@ sidebar_position: 2
 title: Creating your own arguments
 ---
 
-## TODO
+Creating an argument can be done by extending the Argument class which can be imported from the framework:
```
Comment (UndiedGamer) [inline docs/Guide/arguments/creating-your-own-arguments.mdx:6]: Add an example with Args.make too

File: docs/Guide/additional-information/accessing-the-client-in-pieces.mdx [hunk 2]
```diff
@@ -3,4 +3,30 @@ sidebar_position: 0
 title: Accessing the `client` in pieces
```
Comment (favna) [inline docs/Guide/additional-information/accessing-the-client-in-pieces.mdx:3]:
```suggestion
title: Accessing the client in pieces
```

Doesn't render as inline code anyway

File: docs/Guide/additional-information/using-and-extending-container.mdx [hunk 3]
```diff
@@ -3,4 +3,42 @@ sidebar_position: 1
 title: Using and extending `container`
```
Comment (favna) [inline docs/Guide/additional-information/using-and-extending-container.mdx:3]:
```suggestion
title: Using and extending container
```

File: docs/Guide/arguments/creating-your-own-arguments.mdx [hunk 4]
```diff
@@ -3,4 +3,94 @@ sidebar_position: 2
...
+
+```typescript ts2esm2cjs {4|12}
+import { Argument, Resolvers } from '@sapphire/framework';
+import { isTextBasedChannels } from '@sapphire/discord.js-utilities';
+
+export class TextBasedChannelsArgument extends Argument {
```
Comment (favna) [inline docs/Guide/arguments/creating-your-own-arguments.mdx:12]:
```suggestion
export class TextBasedChannelsArgument extends Argument<TextBasedChannelTypes> {
```

File: docs/Guide/arguments/creating-your-own-arguments.mdx [hunk 5]
```diff
@@ -3,4 +3,94 @@ sidebar_position: 2
...
+file:
+
+```typescript ts2esm2cjs {3|15}
+import { Argument, ArgumentContext } from '@sapphire/framework';
+
+interface ArrayArgumentContext extends ArgumentContext {
+  array?: string[];
+}
+
+export class ArrayArgument extends Argument {
```
Comment (favna) [inline docs/Guide/arguments/creating-your-own-arguments.mdx:52]:
```suggestion
export class ArrayArgument extends Argument<string> {
```

File: docs/Guide/arguments/creating-your-own-arguments.mdx [hunk 6]
```diff
@@ -3,4 +3,94 @@ sidebar_position: 2
...
+  }
+}
+
+declare module '@sapphire/framework' {
+  interface ArgType {
+    array: string;
+  }
+}
+```
+
```
Comment (favna) [inline docs/Guide/arguments/creating-your-own-arguments.mdx:71]:
```suggestion
:::info

This is essentially the `enum` argument, one of the [Built-in arguments][bia], which was added in version 2.3.0 of @sapphire/framework

:::
```

Add to the bottom of the doc:
```md
[bia]: ../built-in-arguments
```

File: docs/Guide/arguments/creating-your-own-arguments.mdx [hunk 7]
```diff
@@ -3,4 +3,94 @@ sidebar_position: 2
...
+The above code checks if the array passed in the context includes the parameter, if it does it returns the value, if
+not, an error is thrown. After that, we augment the @sapphire/framework module to include the new argument in the
+ArgType interface (as above).
+
+After that, in your command file:
```
Comment (favna) [inline docs/Guide/arguments/creating-your-own-arguments.mdx:76]:
```suggestion
With all that setup done, you can use the new argument in your command file:
```

**Review by favna — CHANGES_REQUESTED** [review]

File: docs/Guide/arguments/creating-your-own-arguments.mdx [hunk 8]
```diff
@@ -3,4 +3,94 @@ sidebar_position: 2
...
+  }
+}
+```
+
+The above code creates an argument which checks if the provided parameter is a valid text based channel. It first tries
+to resolve the parameter to a channel in the guild, if the the parameter is a valid channel, and the channel is a text
+based channel, it returns the value. If the channel is not a text based channel, or the parameter could not be resolved
+to a channel, it returns an error.
+
+We then augment the @sapphire/framework module to include the new argument in the ArgType interface.
```
Comment (favna) [inline docs/Guide/arguments/creating-your-own-arguments.mdx:37]:
```suggestion
We then augment the `@sapphire/framework` module to include the new argument in the ArgType interface.
```

File: docs/Guide/arguments/creating-your-own-arguments.mdx [hunk 9]
```diff
@@ -3,4 +3,94 @@ sidebar_position: 2
...
+
+declare module '@sapphire/framework' {
+  interface ArgType {
+    array: string;
+  }
+}
+```
+
+The above code checks if the array passed in the context includes the parameter, if it does it returns the value, if
+not, an error is thrown. After that, we augment the @sapphire/framework module to include the new argument in the
```
Comment (favna) [inline docs/Guide/arguments/creating-your-own-arguments.mdx:73]:
```suggestion
not, an error is thrown. After that, we augment the `@sapphire/framework` module to include the new argument in the
```

**Review by favna — CHANGES_REQUESTED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**